### PR TITLE
DBZ-4999 [docs] Documentation of Informix connector

### PR DESCRIPTION
**Pull request by nrkljo**
Documentation of new Informix connector

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/informix.adoc [hunk 1]
```diff
@@ -0,0 +1,2705 @@
...
+----
++
+The command adds a Kafka Connect instance that specifies the name of the image that you created to run your {prodname} connector.
+
+. Create a `KafkaConnector` custom resource that configures your {prodname} Informix connector instance.
++
+You configure a {prodname} Informix connector in a `.yaml` file that specifies the configuration properties for the connector.
+The connector configuration might instruct {prodname} to produce events for a subset of the schemas and tables, or it might set properties so that {prodname} ignores, masks, or truncates values in specified columns that are sensitive, too large, or not needed.
++
+The following example configures a {prodname} connector that connects to a Informix server host, `192.168.99.100`, on port `50000`.
```
Comment (jpechane) [inline documentation/modules/ROOT/pages/connectors/informix.adoc:1910]: Is port number incorrect here?

File: documentation/modules/ROOT/pages/connectors/informix.adoc [hunk 2]
```diff
@@ -0,0 +1,2705 @@
...
+
+* `none` does not apply any adjustment. +
+* `avro` replaces the characters that cannot be used in the Avro type name with underscore. +
+* `avro_unicode` replaces the underscore or characters that cannot be used in the Avro type name with corresponding unicode like _uxxxx. Note: _ is an escape sequence like backslash in Java +
+
+See {link-prefix}:{link-avro-serialization}#avro-naming[Avro naming] for more details.
+|===
+
+[id="informix-advanced-configuration-properties"]
+==== Advanced connector configuration properties
```
Comment (jpechane) [inline documentation/modules/ROOT/pages/connectors/informix.adoc:2337]: I miss the config options like `cdc.timeout`, `cdc.buffersize`.

File: documentation/modules/ROOT/pages/connectors/informix.adoc [hunk 3]
```diff
@@ -0,0 +1,2705 @@
...
+
+// Type: reference
+// ModuleID: monitoring-debezium-informix-connector-schema-history
+// Title: Monitoring {prodname} Informix connector schema history
+[[informix-schema-history-metrics]]
+=== Schema history metrics
+
+include::{partialsdir}/modules/all-connectors/ref-connector-monitoring-schema-history-metrics.adoc[leveloffset=+1]
+
+// Type: assembly
```
Comment (jpechane) [inline documentation/modules/ROOT/pages/connectors/informix.adoc:2658]: Does this really work with the client? I think we need a test similar to `io.debezium.connector.sqlserver.SqlServerChangeTableSetIT` to claim it as suported and verify the procedure.

**Review by jpechane — COMMENTED** [review]
@nrkljo Thanks for the doc, I left couple of comments related to technical side of things.

@roldanbob Could you please take a look at it too? Thanks

File: documentation/modules/ROOT/pages/connectors/informix.adoc [hunk 4]
```diff
@@ -0,0 +1,2705 @@
...
+The Change Data Capture API capture data from databases that have full row logging enabled and captures transactions from the current logical log and processes all transactions sequentially.
+
+The first time a {prodname} Informix connector connects to a Informix database, the connector reads a consistent snapshot of the tables for which the connector is configured to capture changes. By default, this is all non-system tables. There are connector configuration properties that let you specify which tables to put into capture mode, or which tables to exclude from capture mode.
+
+When the snapshot is complete the connector begins emitting change events for committed updates to tables that are in capture mode. By default, change events for a particular table go to a Kafka topic that has the same name as the table. Applications and services consume change events from these topics.
+
+[NOTE]
+====
+The connector requires the use of the Informix Change Streams API for Java, which is packaged as part of the Informix JDBC installation and available on Maven central alongside the latest JDBC drivers.
```
[the remark this answers is not in the document]
Comment (nrkljo) [reply documentation/modules/ROOT/pages/connectors/informix.adoc:42]: I think I expanded this in a similar note, that ended up in a "product" section. Will replace the wording...

File: documentation/modules/ROOT/pages/connectors/informix.adoc [hunk 5]
```diff
@@ -0,0 +1,2705 @@
...
+[id="how-debezium-informix-connectors-read-change-stream-records"]
+=== Change-stream records
+
+After a complete snapshot, when a {prodname} Informix connector starts for the first time, the connector starts consuming change-stream records for the source tables that are in capture mode.
+The connector does the following for each change-data table:
+
+. Reads change events that were created between the last stored, highest LSN and the current, highest LSN.
+. Orders the change events according to the commit LSN and the change LSN for each event.
+  This ensures that the connector emits the change events in the order in which the table changes occurred.
+. Passes commit and change LSNs as offsets to Kafka Connect.
```
[the remark this answers is not in the document]
Comment (nrkljo) [reply documentation/modules/ROOT/pages/connectors/informix.adoc:409]: Yes, will do...

File: documentation/modules/ROOT/pages/connectors/informix.adoc (hunk 1 above)
Comment (nrkljo) [reply documentation/modules/ROOT/pages/connectors/informix.adoc:1910]: Yes, obviously I used the Db2 documentation as a template...

**Comment by roldanbob** [comment]
> @nrkljo Thanks for the doc, I left couple of comments related to technical side of things.
> 
> @roldanbob Could you please take a look at it too? Thanks

I'm about 2/3 through my review. Should finish up on Monday.

**Comment by nrkljo** [comment]
Thanks @roldanbob, I've included your changes (and fixed the commit messages). Looking forward to the rest of your review.
Will work on the metadata issues in the meantime...

**Comment by roldanbob** [comment]
> Thanks @roldanbob, I've included your changes (and fixed the commit messages). Looking forward to the rest of your review. Will work on the metadata issues in the meantime...

Awesome! Thanks for your work, @nrkljo. I'll complete the review of the properties tables today.

File: documentation/modules/ROOT/pages/connectors/informix.adoc [hunk 6]
```diff
@@ -0,0 +1,2486 @@
...
+|Specifies the number of threads that the connector uses when performing an initial snapshot.
+To enable parallel initial snapshots, set the property to a value greater than 1.
+In a parallel initial snapshot, the connector processes multiple tables concurrently.
+ifdef::community[]
+This feature is incubating.
+endif::community[]
+
+|[[informix-property-custom-metric-tags]]<<informix-property-custom-metric-tags, `custom.metric.tags`>>
+|`No default`
+|The custom metric tags will accept key-value pairs to customize the MBean object name which should be appended the end of regular name, each key would represent a tag for the MBean object name, and the corresponding value would be the value of that tag the key is. For example: `k1=v1,k2=v2`.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/informix.adoc:2380]:
```suggestion
|Defines tags for customizing MBean object names by adding metadata that provides contextual information that enables you to organize and categorize metrics data.
Specify a comma-separated list of key-value pairs.
Each key represents a tag for the MBean object name, and the corresponding value represents a value for the key.  +
For example: `k1=v1,k2=v2`.
The connector appends the tag to the base MBean object name.

You can define tags to identify particular application instances, environments, regions, versions, and so forth.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
Completed review of the properties tables. Mostly straighforward changes, but did find that I have questions about a couple of the properties. I posted my question about `topic.cache.size`.  
I also am not sure about the description of the `custom.metric.tags`property.
This was apparently introduced this summer in DBZ-6603, but I'm seeing it for the first time.  I made some changes there, but it might be good if @jpechane and @Naros can take a look at my edits. If my rewrite looks good, I'll apply that to other instances in the connector docs.

**Comment by jpechane** [comment]
As there is no further activity I merged the suggestions and I am going to publish the docs.
@nrkljo @roldanbob Thanks a lot!

**Pull request merged**